zebrad/src/components/mempool.rs

```diff
@@ -30,40 +33,76 @@ pub use self::error::MempoolError;
 #[cfg(test)]
 pub use self::storage::tests::unmined_transactions_in_blocks;
 
+use self::downloads::{
+    Downloads as TxDownloads, GossipedTx, TRANSACTION_DOWNLOAD_TIMEOUT, TRANSACTION_VERIFY_TIMEOUT,
+};
+
+type Outbound = Buffer<BoxService<zn::Request, zn::Response, zn::BoxError>, zn::Request>;
+type State = Buffer<BoxService<zs::Request, zs::Response, zs::BoxError>, zs::Request>;
+type TxVerifier = Buffer<
+    BoxService<transaction::Request, transaction::Response, TransactionError>,
+    transaction::Request,
+>;
+type InboundTxDownloads = TxDownloads<Timeout<Outbound>, Timeout<TxVerifier>, State>;
+
+/// The result of attempting to queue a transaction for downloading/
+/// verifying.
+#[derive(Debug)]
+pub enum DownloadAction {
+    DownloadAction(self::downloads::DownloadAction),
+    InMempool,
+    Rejected,
```

Comment:
These are technically `Error` cases, we may need to create a type that implements `Error` , either download-specific or as a variant of a general `MempoolError` or something. [`zebra-state`](https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/src/error.rs) has some guidance on deriving various error types

Reply:
@upbqdn this might be relevant for when we work on on #2668

Reply:
I agree it's confusing. I'm going to try to revamp the error handling

Reply:
Revamped in 8c29d13

I removed both `DownloadAction`s, moved the error cases into `MempoolError` and changed the methods to return `Result<(), MempoolError>`